packages/http-client/src/ceramic-http-client.ts

```diff
@@ -54,7 +57,7 @@ export default class CeramicClient implements CeramicApi {
    * always have access to the most recent known-about version, without needing
    * to explicitly re-load the document.
    */
-  private readonly _docmap: Record<string, Document>
+  private readonly _docCache: DocCache
```

Comment:
So this is interesting.  Now that we limit the size of the cache here, we limit the behavior described in the comment.  Now it's possible for application code to hold onto a handle to a Document, but stop receiving updates to it once the internal cache fills up and evicts it.  Ideally we'd have a way to know whether a Document handle is still alive outside of our internal uses and prevent evicting it from our internal cache if so.  I'm not actually sure if something like this is possible in javascript though, since there isn't any real concept of a "destructor", or any other way (that I know of) to trigger code to run when an object reference goes out of scope...  @PaulLeCam, do you know if there's any conventional way to solve problems like this in javascript?

Reply:
We discussed this after standup and believe the right way to solve this is to have a single Ceramic-wide EventEmitter for document updates, rather than one per document, that Document instances would subscribe to and only process the events that are relevant to it.